File: examples/php/solarfox-api-request.php
<?php

  // Get cURL resource
  $ch = curl_init();

  // Create body
  $body = json_encode([
    "Headquarters" => [
      "WR-118412E" => [
        "type" => "ELECTRICITY_CONSUMPTION_ENERGY_ABSOLUTE",
        "readings" => [
          [
            "timestamp" => time(),
            "value" => 127932.0
          ]
        ]
      ]
    ]
  ]);

  // Set opts
  curl_setopt_array($ch, [
    CURLOPT_URL => 'http://api.solar-fox.com/import',
    CURLOPT_CUSTOMREQUEST => 'POST',
    CURLOPT_RETURNTRANSFER => 1,
    CURLOPT_HTTPHEADER => ["Content-Type: application/json", "Authorization: test"],
    CURLOPT_POST => 1,
    CURLOPT_POSTFIELDS => $body,
  ]);

  // Send the request & save response to $resp
  $resp = curl_exec($ch);

  if(!$resp) {
    die('Error: "' . curl_error($ch) . '" - Code: ' . curl_errno($ch));
  } else {
    echo "Status: " . curl_getinfo($ch, CURLINFO_HTTP_CODE) . PHP_EOL;
    echo "Body: " . $resp;
  }

  // Close request to clear up some resources
  curl_close($ch);

?>

File: examples/java/SolarfoxAPIRequest.java
import java.io.BufferedReader;
import java.io.DataOutputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class SolarfoxAPIRequest {

  private final String ENDPOINT = "http://api.solar-fox.com/import";

  public static void main(String[] args) throws Exception {
    SolarfoxAPIRequest http = new SolarfoxAPIRequest();
    http.sendPost();
  }

  private void sendPost() throws Exception {
    URL obj = new URL(ENDPOINT);
    HttpURLConnection con = (HttpURLConnection) obj.openConnection();

    // better use simple-json or gson
    String json = "{"
                  + "\"Headquarters\": { "
                  + "   \"WR-118412E\": { "
                  + "    \"type\": \"ELECTRICITY_CONSUMPTION_ENERGY_ABSOLUTE\","
                  + "     \"readings\": [ "
                  + "       {"
                  + "        \"timestamp\": " + System.currentTimeMillis()/1000 + ","
                  + "         \"value\": 128013.0 "
                  + "        }"
                  + "      ]"
                  + "    }"
                  + "  }"
                  + "}";

    String postData = json.toString();

    con.setRequestMethod("POST");
    con.setRequestProperty("Content-Type", "application/json");
    con.setRequestProperty("Authorization","test");
    con.setDoOutput(true);
    DataOutputStream wr = new DataOutputStream(con.getOutputStream());
    wr.writeBytes(postData);
    wr.flush();
    wr.close();

    int responseCode = con.getResponseCode();
    System.out.println("Status: " + responseCode);

    if (responseCode == HttpURLConnection.HTTP_OK) {
      BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
      String inputLine;
      StringBuffer response = new StringBuffer();
      while ((inputLine = in.readLine()) != null) {
          response.append(inputLine);
      }
      in.close();
      System.out.println("Body: " + response.toString());
    } else {
      System.out.println("POST request failed");
    }

  }
}
